src/app/modules/submission/email-submission/email-submission.service.ts

```diff
@@ -86,33 +57,9 @@ const createEmailDataForOneField = (
 export const createEmailData = (
   parsedResponses: ProcessedFieldResponse[],
   hashedFields: Set<string>,
+  authType: AuthType = AuthType.NIL,
 ): EmailData => {
-  // First, get an array of email data for each response
-  // Each field has an array of email data to accommodate table fields,
-  // which have multiple rows of data per field. Hence flatten and maintain
-  // the order of responses.
-  return (
-    parsedResponses
-      .flatMap((response) => createEmailDataForOneField(response, hashedFields))
-      // Then reshape such that autoReplyData, jsonData and formData are each arrays
-      .reduce<EmailData>(
-        (acc, dataForOneField) => {
-          if (dataForOneField.autoReplyData) {
-            acc.autoReplyData.push(dataForOneField.autoReplyData)
-          }
-          if (dataForOneField.jsonData) {
-            acc.jsonData.push(dataForOneField.jsonData)
-          }
-          acc.formData.push(dataForOneField.formData)
-          return acc
-        },
-        {
-          autoReplyData: [],
-          jsonData: [],
-          formData: [],
-        },
-      )
-  )
+  return new SubmissionEmailObj(parsedResponses, hashedFields, authType)
 }
```

Comment:
`createEmailData` is now a dumb wrapper around the `SubmissionEmailObj` constructor and offers no abstraction. shall we get rid of it and just call `new SubmissionEmailObj(...)` directly?

Reply:
Alternatively - keep `createEmailData` but have it return a `Result` for type safety? Either in this PR or the next one.